tsdb/shard.go

```diff
@@ -66,18 +66,39 @@ type Shard struct {
 	LogOutput io.Writer
 }
 
+// ShardLocation is used when calling NewShard to specify the shard's
+// database, retention policy, and location of files on disk.
+type ShardLocation struct {
+	// Name of the database this shard belongs to
+	Database string
+
+	// Name of the retention policy this shard belongs to
+	RetentionPolicy string
+
+	// Path to this shard's location on disk
+	Path string
+
+	// Path to this shard's WAL location
+	WALPath string
+}
+
```

Comment:
I would prefer ShardConfig since it includes other attributes besides just paths. But otherwise lgtm.